testutil/test_helper.go

```diff
@@ -726,3 +727,58 @@ func ParseEvents(out string) ([]*corev1.Event, error) {
 
 	return events, nil
 }
+
+// DiffTestdata returns a diff with the given options
```

Comment:
The comment on the `TestHelper` struct reads: `TestHelper provides helpers for running the linkerd integration tests.` which makes me feel like these unit test helper methods don't really belong in this file.

It also feels unwieldy to me to have this collection of global variables passed in to DiffTestdata everywhere.

Here's what I'd recommend:

Create a new struct in a new file called TestdataDiffer or TestdataHelper or something like that.  Have that struct hold configuration options like `prettyDiff` and `updateFixtures`.  Each `main_test.go` can then declare a single global TestdataHelper with the configuration from the flags.  Individual tests can then call `DiffTestdata` on that global TestdataHelper.

Reply:
Yeah, This makes total sense.

Reply:
Yep good suggestion @adleong